### PR TITLE
feat(rdb): add support for database

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: docs/data-sources/rdb_database.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+layout: "scaleway"
+page_title: "Scaleway: scaleway_rdb_database"
+description: |-
+  Gets information about an RDB database.
+---
+
+# scaleway_rdb_instance
```
Comment (remyleone) [inline docs/data-sources/rdb_database.md:8]:
```suggestion
# scaleway_rdb_database
```

File: docs/data-sources/rdb_database.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Example Usage
+
+```hcl
+# Get the database foobar hosted on instance id 11111111-1111-1111-1111-111111111111
+data "scaleway_rdb_database" "my_db" {
+  instance_id = "11111111-1111-1111-1111-111111111111"
+  name        = "foobar"
+}
+# Find the first database hosted on instance id 11111111-1111-1111-1111-111111111111
```
Comment (remyleone) [inline docs/data-sources/rdb_database.md:20]: Is it not deterministic? Should the name be mandatory?

File: scaleway/resource_rdb_database.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+	if err != nil && !is404Error(err) {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+// Build the resource identifier
+// The resource identifier format is "Region/InstanceId/Database"
```
Comment (remyleone) [inline scaleway/resource_rdb_database.go:147]:
```suggestion
// The resource identifier format is "Region/InstanceId/DatabaseName"
```

File: docs/data-sources/rdb_database.md (hunk 2 above)
Comment (jeansebastienh) [reply docs/data-sources/rdb_database.md:20]: I can make it mandatory sure.

Comment (jeansebastienh) [reply docs/data-sources/rdb_database.md:20]: https://github.com/remyleone/terraform-provider-scaleway/pull/4

File: scaleway/resource_rdb_database.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Database name",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"managed": {
+				Type:        schema.TypeBool,
+				Description: "Whether or notre the database is managed",
```
Comment (jerome-quere) [inline scaleway/resource_rdb_database.go:42]:
```suggestion
				Description: "Whether or not the database is managed",
```

File: scaleway/resource_rdb_database.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+				Description: "User that own the database",
+				Computed:    true,
+			},
+			"size": {
+				Type:        schema.TypeString,
+				Description: "Size of the database",
+				Computed:    true,
+			},
+			// Common
+			"region": regionSchema(),
```
Comment (jerome-quere) [inline scaleway/resource_rdb_database.go:56]: Do we usually include it for nested resources ?

File: scaleway/resource_rdb_database.go [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+		},
+	}
+}
+
+func resourceScalewayRdbDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	rdbAPI, region, err := rdbAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceID := d.Get("instance_id").(string)
```
Comment (jerome-quere) [inline scaleway/resource_rdb_database.go:66]: region should be read from `instance_id`. `instance_id` should be of the form `region/uuid` like all our ids

**Review by jerome-quere — CHANGES_REQUESTED** [review]

File: docs/data-sources/rdb_database.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+}
+```
+
+## Argument Reference
+
+- `instance_id` - (Required) The RDB instance ID.
+
+- `name` - (Required) The name of the RDB instance.
+
+- `region` - (Defaults to [provider](../index.md#region) `region`) The [region](../guides/regions_and_zones.md#zones) in which the RDB instance exists.
```
Comment (jerome-quere) [inline docs/data-sources/rdb_database.md:28]: Should be deduce from `instance_id`

File: scaleway/resource_rdb_database.go (hunk 5 above)
Comment (jeansebastienh) [reply scaleway/resource_rdb_database.go:56]: I would say yes but I don't have the rational : 
```
$ ack "(region|zone)Schema" | grep "scaleway/resource_" -c
25
$ find scaleway -name "resource*" -ls |grep -v _test.go -c
29
```
25 over 29 of our resource include a region or a zone attribute.

https://github.com/scaleway/terraform-provider-scaleway/search?q=regionSchema
https://github.com/scaleway/terraform-provider-scaleway/search?q=zoneSchema

File: scaleway/resource_rdb_database.go (hunk 6 above)
Comment (jeansebastienh) [reply scaleway/resource_rdb_database.go:66]: I'm working on a fix. https://github.com/remyleone/terraform-provider-scaleway/pull/7/

Should I support also non regional id with fallback on the default region?

File: scaleway/resource_rdb_database.go [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceScalewayRdbDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	rdbAPI := newRdbAPI(meta)
+
+	locality, instanceID, err := parseLocalizedID(d.Get("instance_id").(string))
```
Comment (jerome-quere) [inline scaleway/resource_rdb_database.go:62]:
```suggestion
	region, instanceID, err := parseRegionalID(d.Get("instance_id").(string))
```

**Review by jerome-quere — APPROVED** [review]

**Review by jerome-quere — CHANGES_REQUESTED** [review]
see above comment

**Review by jerome-quere — APPROVED** [review]

**Pull request merged**